File: input/java/Person03.java
import java.util.Calendar;
import java.util.GregorianCalendar;
import java.util.Random;

public class Person03 {
    private String firstName;
    private String lastName;
    private Calendar birthday;
    private int favouriteNumber;

    public Person03()
    {
        this.firstName = "";
        this.lastName = "";
        this.birthday = new GregorianCalendar();

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }
    public Person03(String firstName, String lastName)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthday = new GregorianCalendar();

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }
    public Person03(String firstName, String lastName, Calendar birthday)
    {
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthday = birthday;

        Random randomObj = new Random();
        this.favouriteNumber = randomObj.nextInt(42);
    }

    public String fullName()
    {
        return firstName + " " + lastName;
    }

    public int age(Calendar today)
    {
        return today.get(Calendar.YEAR) - birthday.get(Calendar.YEAR);
    }

    public int favNum() {
        return favouriteNumber;
    }
}

File: input/kotlin/Person03.kt
import java.util.GregorianCalendar
import java.util.Calendar
import kotlin.random.Random

class Person03 {
    private var firstName: String
    private var lastName: String
    private var birthday: Calendar
    private var favouriteNumber: Int

    constructor() {
        firstName = ""
        lastName = ""
        birthday = GregorianCalendar()
        favouriteNumber = Random.nextInt(42)
    }

    constructor(firstName: String, lastName: String) {
        this.firstName = firstName
        this.lastName = lastName
        birthday = GregorianCalendar()
        favouriteNumber = Random.nextInt(42)
    }

    constructor(firstName: String, lastName: String, birthday: Calendar) {
        this.firstName = firstName
        this.lastName = lastName
        this.birthday = birthday
        favouriteNumber = Random.nextInt(42)
    }

    fun fullName(): String {
        return "$firstName $lastName"
    }

    fun age(today: Calendar?): Int {
        if (today == null) throw NullPointerException()
        return today[Calendar.YEAR] - birthday[Calendar.YEAR]
    }

    fun favNum(): Int {
        return favouriteNumber
    }
}